tests/functional/specs/control_tags/taxonomy.cy.ts

```diff
@@ -1,59 +1,104 @@
-import { LabelStudio, Taxonomy, Tooltip } from '@heartexlabs/ls-test/helpers/LSF/index';
+import { LabelStudio, Tooltip } from '@heartexlabs/ls-test/helpers/LSF/index';
+import { useTaxonomy } from '@heartexlabs/ls-test/helpers/LSF';
 import {
   dataWithPrediction,
   simpleData,
   taxonomyConfig,
-  taxonomyConfigWithMaxUsages
+  taxonomyConfigWithMaxUsages,
+  taxonomyResultWithAlias
 } from '../../data/control_tags/taxonomy';
+import { FF_PROD_309, FF_TAXONOMY_ASYNC, FF_TAXONOMY_LABELING } from '../../../../src/utils/feature-flags';
 
-describe('Control Tags - Taxonomy', () => {
-  it('should show hint for <Choice />', () => {
-    LabelStudio.params()
-      .config(taxonomyConfig)
-      .data(simpleData)
-      .withResult([])
-      .init();
-
-    Taxonomy.open();
-    Taxonomy.findItem('Choice 2').trigger('mouseenter');
-    Tooltip.hasText('A hint for Choice 2');
-  });
+const taxonomies = {
+  'Old Taxonomy': useTaxonomy('&:eq(0)'),
+  'New Taxonomy': useTaxonomy('&:eq(0)', true),
+};
 
-  it('should show error message if there are more choices selected than maxUsages is set', () => {
-    LabelStudio.init({
-      config: taxonomyConfigWithMaxUsages,
-      task: dataWithPrediction,
+Object.entries(taxonomies).forEach(([title, Taxonomy]) => {
+  describe('Control Tags - ' + title, () => {
+    beforeEach(() => {
+      if (Taxonomy.isNew) {
+        LabelStudio.addFeatureFlagsOnPageLoad({
+          [FF_TAXONOMY_ASYNC]: true,
+          [FF_TAXONOMY_LABELING]: true,
+        });
+      }
     });
 
-    cy.contains('button', 'Update').click();
+    it('should show hint for <Choice />', () => {
+      LabelStudio.addFeatureFlagsOnPageLoad({
+        [FF_PROD_309]: true,
```

Comment:
For now It's already set globally. But it's just in case if that was not known. I mean I'm not against doing it twice to be sure.
https://github.com/HumanSignal/label-studio-frontend/blob/e888b2df9e16dbffb39f39529fa7de9dba0d92c7/tests/functional/feature-flags.ts#L3-L15